var/spack/repos/builtin/packages/py-jupyter-core/package.py

```diff
@@ -10,8 +10,9 @@ class PyJupyterCore(PythonPackage):
     """Core Jupyter functionality"""
 
     homepage = "http://jupyter-core.readthedocs.io/"
-    url      = "https://github.com/jupyter/jupyter_core/archive/4.2.0.tar.gz"
+    url      = "https://pypi.io/packages/source/j/jupyter-core/jupyter_core-1.17.2.tar.gz"
```

Comment:
I'm all for switching the download URL to PyPI, but you'll need to update all previous checksums and make sure that the version in the URL is present in the package:
```suggestion
    url      = "https://pypi.io/packages/source/j/jupyter-core/jupyter_core-4.6.0.tar.gz"
```